File: backend/src/main/java/ca/metricalsky/winston/entity/view/AuthorVideoView.java
package ca.metricalsky.winston.entity.view;

import java.time.OffsetDateTime;

public interface AuthorVideoView {

    String getVideoId();

    String getVideoTitle();

    String getVideoThumbnailUrl();

    Long getTotalCommentCount();

    Long getReplyCount();

    OffsetDateTime getFirstCommentedAt();

    OffsetDateTime getLastCommentedAt();

    default Long getCommentCount() {
        return getTotalCommentCount() - getReplyCount();
    }
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/view/AuthorVideoView.kt
package ca.metricalsky.winston.database.view

import java.time.OffsetDateTime

interface AuthorVideoView {

    fun getVideoId(): String

    fun getVideoTitle(): String

    fun getVideoThumbnailUrl(): String

    fun getTotalCommentCount(): Long

    fun getReplyCount(): Long

    fun getFirstCommentedAt(): OffsetDateTime

    fun getLastCommentedAt(): OffsetDateTime

    fun getCommentCount(): Long {
        return getTotalCommentCount() - getReplyCount()
    }
}
